pages/cart.tsx

```diff
@@ -21,123 +22,88 @@ export async function getStaticProps({
-      currencyCode: data.currency.code,
-    }
-  )
-  const { price: total } = usePrice(
-    data && {
-      amount: data.cart_amount,
-      currencyCode: data.currency.code,
-    }
-  )
-
-  const items = data?.line_items.physical_items ?? []
-
-  const error = null
-  const success = null
+  const {
+    items,
+    isLoading,
+    isError,
+    isEmpty,
+    subtotal,
+    total,
+    currency,
+  } = useCart()
 
   return (
-    <div className="grid lg:grid-cols-12">
-      <div className="lg:col-span-8">
-        {isEmpty ? (
-          <div className="flex-1 px-12 py-24 flex flex-col justify-center items-center ">
-            <span className="border border-dashed border-secondary rounded-full flex items-center justify-center w-16 h-16 bg-primary p-12 rounded-lg text-primary">
-              <Bag className="absolute" />
-            </span>
-            <h2 className="pt-6 text-2xl font-bold tracking-wide text-center">
-              Your cart is empty
-            </h2>
-            <p className="text-accents-6 px-10 text-center pt-2">
-              Biscuit oat cake wafer icing ice cream tiramisu pudding cupcake.
-            </p>
-          </div>
-        ) : error ? (
-          <div className="flex-1 px-4 flex flex-col justify-center items-center">
-            <span className="border border-white rounded-full flex items-center justify-center w-16 h-16">
-              <Cross width={24} height={24} />
-            </span>
-            <h2 className="pt-6 text-xl font-light text-center">
-              We couldn’t process the purchase. Please check your card
-              information and try again.
-            </h2>
-          </div>
-        ) : success ? (
-          <div className="flex-1 px-4 flex flex-col justify-center items-center">
-            <span className="border border-white rounded-full flex items-center justify-center w-16 h-16">
-              <Check />
-            </span>
-            <h2 className="pt-6 text-xl font-light text-center">
-              Thank you for your order.
-            </h2>
-          </div>
-        ) : (
-          <div className="px-4 sm:px-6 flex-1">
-            <Text variant="pageHeading">My Cart</Text>
-            <Text variant="sectionHeading">Review your Order</Text>
-            <ul className="py-6 space-y-6 sm:py-0 sm:space-y-0 sm:divide-y sm:divide-accents-2 border-b border-accents-2">
-              {items.map((item) => (
-                <CartItem
-                  key={item.id}
-                  item={item}
-                  currencyCode={data?.currency.code!}
-                />
-              ))}
-            </ul>
-            <div className="my-6">
-              <Text>
+    <div className="px-4 pt-2 sm:px-6 flex-1">
+      <Text variant="pageHeading">My Cart</Text>
+      {isLoading ? (
+        <CartSkeleton />
+      ) : isError ? (
+        <div className="mt-2">Failed to load</div>
+      ) : isEmpty ? (
+        <CartEmpty />
+      ) : (
```

Comment:
If you handle `isEmpty` first, you don't have to check for `data === undefined`, only for `!data`. Also handle the error first (`data` is a getter)

Reply:
I don't follow what you mean @lfades, care to expand? Thanks!

Reply:
@lfades by the way, `isEmpty` starts being true, then it goest to false while loading, and then it comes back to the proper value after fetch is done. So I don't think I can fully rely on it to determine if the cart is loading.

Reply:
This: https://github.com/vercel/commerce/pull/33#issuecomment-718966924 <- The order I used there actually makes a difference.

Reply:
![Screen Shot 2020-10-30 at 10 41 36 PM](https://user-images.githubusercontent.com/5664/97768365-49159300-1b01-11eb-82aa-f6ce7231a43b.png)

That's what happens when I load `/cart/` directly. If I follow the order you proposed, I get a flash of empty cart no matter what because of `isEmpty` being initially true.

Reply:
I just put the logic back into the page component and moved error to the top of the rendering logic, but I still keep a custom `isLoading` flag because I don't think `isEmpty` is usable on the initial render. Please take a look at the update.

Reply:
@marbiano You're right, there must be something we can do in the hook itself then 🤔

Reply:
@lfades maybe we can live with this solution until the hook is updated by the BC team?

Reply:
I can make the update (I did those hooks, we moved them to a BC repo before releasing the example), basically we want a `isLoading` prop alongside the `isEmpty` one? makes sense 🤔

Reply:
Oh I didn't know that! Yeah, `isLoading` would be perfect and it would also mean almost no update on a potential `react-query` alternate implementation. 😁